### PR TITLE
Feature/ipa production plant improvement

### PR DESCRIPTION
**Pull request by ipa-srd-rd**
In this PR the new environment ipa-production-plant has been updated. Objects like a Reflector Marker or conveyor were added to cob_gazebo_objects because they were used in this environment. 
Further the spawn_object script was modified such that it can spawn now models from the database or sdf's and that it's possible to specify rooms for the objects in cob_environments. With that it's easily possible to spawn objects just for one or more rooms. 

This PR is related to  https://github.com/ipa320/cob_environments/pull/105

### CODE REVIEW

**Comment by stefandoerr** [comment]
@ipa-nhg could you review this PR? @ipa-srd-rd will only be here at IPA till the end of july. Should be merged by then.

File: cob_bringup_sim/scripts/spawn_model.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/python
```
Comment (floweisshardt) [inline cob_bringup_sim/scripts/spawn_model.py:1]: what does spawn_model do different than spawn_object? Can't this be integrated into one spawn script?

File: cob_bringup_sim/scripts/spawn_object.py [hunk 2]
```diff
@@ -61,19 +61,26 @@
...
 
-from gazebo_msgs.srv import SpawnModel, SpawnModelRequest, DeleteModel, DeleteModelRequest
+from spawn_model import SpawnModel
 from geometry_msgs.msg import Pose
 import tf.transformations as tft
 
 parents = {}
-compound_keys={}
+compound_keys = {}
+visualizedRooms = {}
```
Comment (floweisshardt) [inline cob_bringup_sim/scripts/spawn_object.py:70]: spawn_objects didn't know about "rooms" so far. This seems to be an either project specific or environment specific extension, so I like to NOT merge this.

File: cob_bringup_sim/scripts/spawn_object.py [hunk 3]
```diff
@@ -155,89 +171,32 @@ def get_flat_dict(objects, parent_name):
...
-            continue
+            newModel.urdf_format = True
+            newModel.file_name = roslib.packages.get_pkg_dir('cob_gazebo_objects') + '/objects/' + model_string+ '.' + model_type
 
+        # ToDo: Implement same for urdf.xacro
+        # elif model_type == "urdf.xacro":
+        #     p = os.popen("rosrun xacro xacro.py " + file_location)
 
-        # check if object is already spawned
-        srv_delete_model = rospy.ServiceProxy('gazebo/delete_model', DeleteModel) # TODO this service causes gazebo (current groovy version) to crash
```
Comment (floweisshardt) [inline cob_bringup_sim/scripts/spawn_object.py:217]: please do not delete the `delete_model` functionality

File: cob_bringup_sim/scripts/spawn_object.py [hunk 4]
```diff
@@ -155,89 +171,32 @@ def get_flat_dict(objects, parent_name):
...
 
-        if exists:
-            rospy.loginfo("Model %s already exists in gazebo. Model will be updated.", key)
-
-        # spawn new model
-        req = SpawnModelRequest()
-        req.model_name = key # model name from command line input
-        req.model_xml = xml_string
-        req.initial_pose = object_pose
+        elif model_type == "database":
```
Comment (floweisshardt) [inline cob_bringup_sim/scripts/spawn_object.py:193]: what does model_type="database" mean?

**Comment by stefandoerr** [comment]
@ipa-srd-rd could you integrate/respond to the review comments?

**Comment by mgruhler** [comment]
Do Not Merge: I guess this PR is deprecated and can be closed, as the functionality is included in #114 

@ipa-srd-rd Please elaborate...

**Comment by ipa-srd-rd** [comment]
The functionality of this PR is included in #114 and will therefore be closed.

**Pull request closed without merging**